store/sqlstore/user_store.go

```diff
@@ -1361,3 +1361,466 @@ func (us SqlUserStore) InferSystemInstallDate() store.StoreChannel {
+					            ON ParentPostAuthor.Id = (SELECT UserId FROM Posts ParentPost WHERE ParentPost.Id = Posts.ParentId)
+
+					            JOIN Channels
+					            ON Channels.Id = Posts.ChannelId
+
+											JOIN ChannelMembers
+					            ON ChannelMembers.ChannelId = Channels.Id
+					            AND ChannelMembers.UserId = :uid
+
+					        WHERE Channels.Type = 'G'
+					        AND Posts.OriginalId = ''
+				    		) TargetInteraction
+				    ON TargetInteraction.ChannelId = ChannelMembers.ChannelId
+
+				WHERE (TargetInteraction.Interaction_Type = 'DirectMessage' -- direct message from someone other than current user in group channel
+				        AND TargetInteraction.UserId != :uid
+				        AND ChannelMembers.UserId = :uid)
+
+				OR  (TargetInteraction.Interaction_Type = 'DirectMessage' -- direct message from current user in group channel
+				    AND TargetInteraction.UserId = :uid
+				    AND ChannelMembers.UserId != :uid)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, not written by current user, to a post that the current user IS an author of, in a group channel
+				        AND TargetInteraction.UserId != :uid
+				        AND Parent_Post_Author_Id = :uid
+				        AND ChannelMembersUsers.Id = :uid) -- just want one user returned from channel members (the current user) (will record as a reply)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, not written by current user, to a post that the current user IS NOT an author of, in a group channel
+				        AND TargetInteraction.UserId != :uid
+				        AND Parent_Post_Author_Id != :uid
+				        AND ChannelMembersUsers.Id = :uid) -- just want one user returned from channel members (the current user) (will record as a DM)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, written by current user, to a post that the current user IS NOT an author of, in a group channel
+				        AND TargetInteraction.UserId = :uid
+				        AND Parent_Post_Author_Id != :uid
+				        AND ChannelMembersUsers.Id != :uid)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, written by current user, to a post that the current user IS an author of, in a group channel
+				        AND TargetInteraction.UserId = :uid
+				        AND Parent_Post_Author_Id = :uid
+				        AND ChannelMembersUsers.Id != :uid)
+				)
+			) AS UserInteractions
+
+			WHERE Username IN (SELECT Username -- all interactions must take place with users on your team
+			                          FROM Users
+
+			                            JOIN TeamMembers
+			                            ON TeamMembers.TeamID = :teamId
+			                            AND TeamMembers.UserId = Users.Id
+			                        )
+			`
+			var userInteractions []*model.UserInteraction
+
+			if _, err :=  us.GetReplica().Select(&userInteractions, query, map[string]interface{}{"uid": uid, "teamId": teamId}); err != nil {
```

Comment:
Passing the address of an array here seems like a funny thing to do to me. Presumably this is so that the function can mutate the variable to point at an array that it will allocate? I assume this is fine in Go, but it would be a strange idiom in most C-family languages, so maybe double check.